internal/graph/check_test.go

```diff
@@ -3072,3 +3076,992 @@ func TestConsumeUsersets(t *testing.T) {
+			},
+			expectedDispatches: []dispatchMsgAssertion{
+				{
+					err:          fmt.Errorf("condition not found"),
+					shortCircuit: false,
+					dispatch:     nil,
+				},
+			},
+		},
+		{
+			name: "good_condition_wildcard",
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKeyWithCondition("group:1", "member", "user:*", "condition1", nil),
+				tuple.NewTupleKeyWithCondition("group:1", "member", "group:2#member", "condition1", nil),
+			},
+			expectedDispatches: []dispatchMsgAssertion{
+				{
+					err:          nil,
+					shortCircuit: true,
+					dispatch:     nil,
+				},
+			},
+		},
+		{
+			name: "good_condition_non_wildcard",
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKeyWithCondition("group:1", "member", "group:2#member", "condition1", nil),
+			},
+			expectedDispatches: []dispatchMsgAssertion{
+				{
+					err:          nil,
+					shortCircuit: false,
+					dispatch:     tuple.NewTupleKey("group:2", "member", "user:maria"),
+				},
+			},
+		},
+		{
+			name: "multiple_tuples",
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKeyWithCondition("group:1", "member", "group:2#member", "condition1", nil),
+				tuple.NewTupleKeyWithCondition("group:1", "member", "group:3#member", "condition1", nil),
+			},
+			expectedDispatches: []dispatchMsgAssertion{
+				{
+					err:          nil,
+					shortCircuit: false,
+					dispatch:     tuple.NewTupleKey("group:2", "member", "user:maria"),
+				},
+				{
+					err:          nil,
+					shortCircuit: false,
+					dispatch:     tuple.NewTupleKey("group:3", "member", "user:maria"),
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+			iter := storage.NewConditionsFilteredTupleKeyIterator(storage.NewStaticTupleKeyIterator(tt.tuples), filter)
+			checker := NewLocalChecker()
+			defer checker.Close()
+			mockResolver := NewMockCheckResolver(ctrl)
+			checker.SetDelegate(mockResolver)
+
+			req := &ResolveCheckRequest{
+				TupleKey:        tuple.NewTupleKeyWithCondition("document:doc1", "viewer", "user:maria", "condition1", nil),
+				RequestMetadata: NewCheckRequestMetadata(20),
+			}
+
+			var expectedMsgDispatches []*openfgav1.TupleKey
+
+			for _, expectedMsg := range tt.expectedDispatches {
+				if expectedMsg.dispatch != nil {
+					expectedMsgDispatches = append(expectedMsgDispatches, expectedMsg.dispatch)
+				}
+			}
+
+			if len(expectedMsgDispatches) > 0 {
+				mockResolver.EXPECT().ResolveCheck(gomock.Any(), gomock.AssignableToTypeOf(req)).Times(len(expectedMsgDispatches)).DoAndReturn(
+					func(_ context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error) {
+						require.NotEmpty(t, req)
+						var item *openfgav1.TupleKey
+						item, expectedMsgDispatches = expectedMsgDispatches[0], expectedMsgDispatches[1:]
+						require.Equal(t, item, req.GetTupleKey())
+						return nil, nil
+					})
+			}
```

Comment:
This can be simplified:

```go
for _, expectedMsg := range tt.expectedDispatches {
				if expectedMsg.dispatch != nil {
					mockResolver.EXPECT().ResolveCheck(gomock.Any(), gomock.Any()).Times(1).DoAndReturn(
						func(_ context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error) {
							require.NotEmpty(t, req)
							require.Equal(t, expectedMsg.dispatch, req.GetTupleKey())
							return nil, nil
						})
				}
			}
```